src/main/java/com/esri/samples/raster/raster_layer_file/RasterLayerFileSample.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.raster.raster_layer_file;
+
+import java.io.File;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.Raster;
+
+public class RasterLayerFileSample extends Application {
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Raster Layer File");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a raster from a local raster file
+      Raster raster = new Raster(new File("./samples-data/raster/Shasta.tif").getAbsolutePath());
+
+      // create a raster layer
+      RasterLayer rasterLayer = new RasterLayer(raster);
+
+      // create a Map with imagery basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+
+      // add the map to a map view
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add the raster as an operational layer
+      map.getOperationalLayers().add(rasterLayer);
+
+      // set viewpoint on the raster
+      rasterLayer.addDoneLoadingListener(() -> mapView.setViewpointGeometryAsync(rasterLayer.getFullExtent()));
```

Comment:
I like it better when it  is a little zoomed out. `mapView.setViewpointGeometryAsync(rasterLayer.getFullExtent(), 200);`